packages/apollo-server-core/src/types.ts

```diff
@@ -64,6 +65,25 @@ type BaseConfig = Pick<
   | 'cache'
 >;
 
+export type Unsubscriber = () => void;
+export type SchemaChangeCallback = (schema: GraphQLSchema) => void;
+
+export type GraphQLServiceConfig = {
+  schema: GraphQLSchema;
+  executor: GraphQLExecutor;
+};
+
+export type EngineConfig = {
```

Comment:
I'm afraid that this `EngineConfig` might be confused with "the configuration which goes into the `engine` property of the `ApolloServer` constructor `config`.  Thoughts about `GatewayEngineConfig`?

Reply:
`GraphQLServiceEngineConfig`? As gateway isn't mentioned in AS core at all.